src/main/java/com/maniek/medicalclinic/model/entity/Doctor.java

```diff
@@ -0,0 +1,43 @@
+package com.maniek.medicalclinic.model.entity;
+
+import com.maniek.medicalclinic.model.Role;
+import com.maniek.medicalclinic.model.Specialization;
+import com.maniek.medicalclinic.model.dto.DoctorDTO;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
```

Comment:
Adnotacja @Data dodaje nam w szczegolnosci @EqualsAndHashCode oraz @ToString.
W przypadku gdy my mamy Zdefinowane dwustronne mapowanie tj. Doktor ma List<Facility> oraz Facility ma List<Doctor> to dochodzi nam do problemu. 
Przeanalizujmy problem:
Mamy Doktor1, Doktor2, Doktor3 I mamy Facility1, Facility2,Facility3
Zalozmy, ze Doktor1 jest przypisany do Facility1. 
Defaultowa implementacja z lomboka @ToString wypisuje nam wszystkie pola. 
Wypisze nam wiec firstName, lastName, specialization i sprobuje wypisac facilities.
Probujac wypisac facilities wchodzi do klasy Facility i wypisuje wszystkie pola w szczegolnosci doctors (Liste doktorow). Wiec wchodzi do doktora i probuje wypisac wszystkie pola w szczegolnosci facilities i progrma nam sie zapetla bo probujac wypisac doktora -> probuje wypisac facility -> a proboujac wypisac facility znowu probuje wypisac doctora. 
doctor.toString() -> facility.toString() -> doctor.toString() ---> .. -> az zabraknie pamieci w aplikacji i sie wywali. W takiej sytuacji mozemy np zdefiniowca wlasna metode @ToString, ktora zamiast wypisac listeFacilities wypisez liste tylko idkow. 
Mamy rowniez podobny problem z equalsAndHashCode

Reply:
Nalezy wiec dodac samemu odpowiednie implementacje equals, hashcode i toString